src/bootstrap/src/core/build_steps/format.rs

```diff
@@ -114,6 +114,9 @@ fn print_paths(verb: &str, adjective: Option<&str>, paths: &[String]) {
     } else {
         println!("fmt: {verb} {len} {adjective}files");
     }
+    if len > 1000 {
+        println!("hint: if this number seems too high, try running `git fetch origin master");
+    }
```

Comment:
I think the if here should include "not a CI" check as we are not interested on this for CI environments.